File: test/language/templates/java/templates/StructLongTemplateArgumentTest.java
package templates;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import zserio.runtime.io.BitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamWriter;

import templates.struct_long_template_argument.StructLongTemplateArgument;
import templates.struct_long_template_argument.ThisIsVeryVeryVeryLongNamedStructure;
import templates.struct_long_template_argument.TemplatedStruct_ThisIsVeryVeryVeryLongNamedStructure_ThisIsVery_;

public class StructLongTemplateArgumentTest
{
    @Test
    public void readWrite() throws IOException
    {
        final TemplatedStruct_ThisIsVeryVeryVeryLongNamedStructure_ThisIsVery_ templ =
                new TemplatedStruct_ThisIsVeryVeryVeryLongNamedStructure_ThisIsVery_(
                        new ThisIsVeryVeryVeryLongNamedStructure("StringT"),
                        new ThisIsVeryVeryVeryLongNamedStructure("StringU"),
                        new ThisIsVeryVeryVeryLongNamedStructure("StringV"));
        final StructLongTemplateArgument structLongTemplateArgument = new StructLongTemplateArgument(templ);

        final ByteArrayBitStreamWriter writer = new ByteArrayBitStreamWriter();
        structLongTemplateArgument.write(writer);

        final BitStreamReader reader = new ByteArrayBitStreamReader(
                writer.toByteArray(), writer.getBitPosition());
        final StructLongTemplateArgument readStructLongTemplateArgument =
                new StructLongTemplateArgument(reader);
        assertTrue(structLongTemplateArgument.equals(readStructLongTemplateArgument));
    }
}

File: test/language/templates/cpp98/StructLongTemplateArgumentTest.cpp
#include "gtest/gtest.h"

#include "templates/struct_long_template_argument/StructLongTemplateArgument.h"

namespace templates
{
namespace struct_long_template_argument
{

TEST(StructLongTemplateArgumentTest, readWrite)
{
    ThisIsVeryVeryVeryLongNamedStructure field1;
    field1.setData("StringT");
    ThisIsVeryVeryVeryLongNamedStructure field2;
    field2.setData("StringU");
    ThisIsVeryVeryVeryLongNamedStructure field3;
    field3.setData("StringV");
    TemplatedStruct_ThisIsVeryVeryVeryLongNamedStructure_ThisIsVery_ templ;
    templ.setField1(field1);
    templ.setField1(field2);
    templ.setField1(field3);
    StructLongTemplateArgument structLongTemplateArgument;
    structLongTemplateArgument.setStructNameOverflow(templ);

    zserio::BitStreamWriter writer;
    structLongTemplateArgument.write(writer);
    size_t bufferSize = 0;
    const uint8_t* buffer = writer.getWriteBuffer(bufferSize);

    zserio::BitStreamReader reader(buffer, bufferSize);
    StructLongTemplateArgument readStructLongTemplateArgument(reader);

    ASSERT_TRUE(structLongTemplateArgument == readStructLongTemplateArgument);
}

} // namespace struct_long_template_argument
} // namespace templates
